generate_network.py

```diff
@@ -0,0 +1,155 @@
+#! /usr/bin/env python3
+
+import matplotlib.pyplot as mp
+import networkx as nx
+import networkx.readwrite.json_graph
+import random, math
+import json
+import argparse,sys
+
+def write_json(filename, data_2_dump):
+    json.dump(data_2_dump, open(filename,'w'), indent=2)
+
+# has trouble with non-integer/non-hashable keys 
+def read_json(filename):
+    with open(filename) as f:
+        jdata = json.load(f)
+    return nx.node_link_graph(jdata)
+
+def draw(H):
+    nx.draw(H, pos=nx.kamada_kawai_layout(H), with_labels=True)
+    mp.show()
+    mp.savefig("topology.png", format="PNG")
+
+def init_arg_parser() :
+    # Initialize parser, add arguments and set the defaults
+    parser = argparse.ArgumentParser(
+            prog = 'generate_network',
+            description = '''Generates and outputs 
+            the Waku network conforming to input parameters''',
+            epilog = '''The defaults are: -o "Topology.json"; 
+            -n 1;  -t 1;  -p 1; -T "configuration_model"''')
+    parser.add_argument("-o", "--output", 
+            default='Topology.json', dest='fname', 
+            help='output json filename for the Waku network', 
+            type=str, metavar='<file_name>')
+    parser.add_argument("-n", "--numnodes", 
+            default=1, dest='num_nodes', 
+            help='number of nodes in the Waku network', 
+            type=int, metavar='<#nodes>')
+    parser.add_argument("-t", "--numtopics", 
+            default=1, dest='num_topics', 
+            help='number of topics in the Waku network', 
+            type=int, metavar='<#topics>')
+    parser.add_argument("-T", "--type", 
```

Comment:
Not sure what this argument is doing

Reply:
-T allows us to specify the total number of topics in the Waku network and each  Waku node handles a random subset of these topics.